odoo/orm/domains.py

```diff
@@ -924,6 +921,61 @@ def _optimize_for_sql(self, model: BaseModel) -> Domain:
         object.__setattr__(self, '_opt_level', OptimizationLevel.FOR_SQL)
         return self
 
+    def _optimize_field_search_method(self, model: BaseModel) -> Domain:
+        field = self._field(model)
+        operator, value = self.operator, self.value
+        # simplify further for boolean fields
+        if field.type == 'boolean':
+            # special case to ease implementation: always use (field_expr, 'in'/'not in', [True])
```

Comment:
- This normalization is not intuitive to me.  Please explain the motivation.  What about always using `in` instead, so that all non-trivial conditions on operators end up as either `x in [True]` or `x in [False]`?
- Turn that normalization into an optimization, as it may be useful everywhere.